equinox/nn/pool.py

```diff
@@ -0,0 +1,89 @@
+from typing import Sequence, Tuple, Union
+
+import jax.lax as lax
+import numpy as np
+
+from ..custom_types import Array
+from ..module import Module, static_field
+
+
+class Pool(Module):
+    operation: str = static_field()
+    num_spatial_dims: int = static_field()
+    kernel_size: Union[int, Sequence[int]] = static_field()
+    stride: Sequence[int] = static_field()
+    padding: Union[str, Sequence[Tuple[int, int]]] = static_field()
+
+    def __init__(
+        self,
+        operation: str,
+        num_spatial_dims: int,
+        kernel_size: Union[int, Sequence[int]],
+        stride: Sequence[int] = None,
+        padding: Union[str, Sequence[Tuple[int, int]]] = "SAME",
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        self.operation = operation
+        self.num_spatial_dims = num_spatial_dims
+        self.kernel_size = kernel_size
+        self.stride = stride
+        if stride is None:
+            self.stride = self.kernel_size
+        self.padding = padding
+
+    def __call__(self, x: Array) -> Array:
+        if self.operation == "max":
+            output = lax.reduce_window(
+                x,
+                0.0,
+                lax.max,
+                (1,) + self.kernel_size,
+                (1,) + self.stride,
+                self.padding,
+            )
+            # print(x)
+        elif self.operation == "avg":
+            output = lax.reduce_window(
+                x,
+                0.0,
+                lax.add,
+                (1,) + self.kernel_size,
+                (1,) + self.stride,
+                self.padding,
+            ) / np.prod(self.kernel_size)
+
+        return output
+
+
+class Pool2d_Avg(Pool):
```

Comment:
The appropriate naming convention here would be `AvgPool2d`. (In particular it's generally common in Python to avoid using underscores in class names.)

I think if we do this we should probably add the full gamut of `{Avg,Max}Pool{1,2,3}d`.